blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/ScmResourceImpl.java

```diff
@@ -59,7 +59,7 @@ public Link getLink() {
     }
 
     private @Nonnull User checkPermission(){
-        ACL acl = Jenkins.getInstance().getACL();
+        ACL acl = item.getACL();
```

Comment:
I'm not sure about any ACL test here.  The permission IIUC is controlled by the underlying SCM (e.g. github) using a user based ket/token.
Thus an admin user may be in chanrge of creating the multibranch pipelines - but if a user can push to github why not allow the user to use BO to do it even if they can not configure the job?

This is certainly better than it was so it gets an approval from me - but seems questionable in the bigger scheme of things.
/cc @vivek @michaelneale

Reply:
@jtnord This was added as extra check as editing a Jenkinsfile on existing job was considered job configuration and that user is operating in terms of existing job and editing pipeline was attached to configure permission. Main idea behind this was to reject edit of pipeline for users without configure permission. We can relax this permission if needed, but thats anyways is orthogonal to this PR.

Reply:
yeah its all a bit odd to me - but then again, a "root" user that can be not allowed to access a folder (which is what alvaro was catering for) kind of also does my head in, so I am not sure I really understand what is going on. 

As you can configure a lot with the Jenkinsfile, it does make sense to trust the SCM as the source of access truth...